### PR TITLE
Allowing Restart=on-failure for Type=oneshot

### PR DESCRIPTION
**Pull request by robermorales**
It partially solves #2582

### CODE REVIEW

File: src/core/service.c [hunk 1]
```diff
@@ -514,8 +514,8 @@ static int service_verify(Service *s) {
                 return -EINVAL;
         }
 
-        if (s->type == SERVICE_ONESHOT && s->restart != SERVICE_RESTART_NO) {
-                log_unit_error(UNIT(s), "Service has Restart= setting other than no, which isn't allowed for Type=oneshot services. Refusing.");
+        if (s->type == SERVICE_ONESHOT && (s->restart != SERVICE_RESTART_NO && s->restart != SERVICE_RESTART_ON_FAILURE)) {
```
Comment (keszybz) [inline src/core/service.c:517]: `!IN_SET(s->restart, SERVICE_RESTART_NO, SERVICE_RESTART_ON_FAILURE)`

File: src/core/service.c [hunk 2]
```diff
@@ -514,8 +514,8 @@ static int service_verify(Service *s) {
...
+                log_unit_error(UNIT(s), "Service has Restart= setting other than no or on-failure, which are the only allowed for Type=oneshot services. Refusing.");
```
Comment (keszybz) [inline src/core/service.c:518]: There's a word missing in that sentence:
> which are the only [WHAT] allowed 

Maybe "Illegal Restart= setting for service with Type=oneshot; only \"no\" and \"on-failure\" are allowed. Ignoring."?

**Comment by poettering** [comment]
well, if we open this up we should probably open it up for all restart types, that conceptually can be made fit. 

quite frankly, i don't really see all the implications of this change. Let's please make sure this actually works properly: could you add a test case in test/TEST-… for this?

File: src/core/service.c (hunk 1 above)
Comment (robermorales) [reply src/core/service.c:517]: thanks for the tip

File: src/core/service.c (hunk 2 above)
Comment (robermorales) [reply src/core/service.c:518]: true.

**Comment by robermorales** [comment]
I will work on some tests and think about the rest of types.

**Comment by poettering** [comment]
I still think adding this would be good, but given the PR stalled, let's close this.

**Pull request closed without merging**